Create TubeGeometry from LineCurve3: build error

HI!
I'm trying to create TubeGeometry from LineCurve3.
```
std::vector<threepp::Vector3> points;
points.push_back({ 0, 0, 0 });
points.push_back({ 1, 1, 1 });
auto curve = std::make_shared<threepp::LineCurve3>(points[0], points[1]);
auto tube = threepp::TubeGeometry::create(curve);
```
I've got a building error:
```
[build] XXX/test.cpp: In function ‘void test()’:
[build] XXX/test.cpp:65:54: error: no matching function for call to ‘threepp::TubeGeometry::create(std::shared_ptr<threepp::LineCurveT<threepp::Vector3> >&)’
[build]    65 |             auto tube = threepp::TubeGeometry::create(curve);
[build]       |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
```
But , there is no building error when I'm trying to create TubeGeometry from CatmullRomCurve3.

--- a/include/threepp/extras/core/Curve.hpp
+++ b/include/threepp/extras/core/Curve.hpp
@@ -91,21 +91,17 @@ namespace threepp {
         mutable std::vector<float> cacheArcLengths;
     };
 
+    struct FrenetFrames {
+
+        std::vector<Vector3> tangents;
+        std::vector<Vector3> normals;
+        std::vector<Vector3> binormals;
+
+        static FrenetFrames compute(const Curve<Vector3>& curve, unsigned int segments, bool closed);
+    };
 
     typedef Curve<Vector2> Curve2;
-
-    class Curve3: public Curve<Vector3> {
-
-    public:
-        struct FrenetFrames {
-
-            std::vector<Vector3> tangents;
-            std::vector<Vector3> normals;
-            std::vector<Vector3> binormals;
-        };
-
-        FrenetFrames computeFrenetFrames(unsigned int segments, bool closed);
-    };
+    typedef Curve<Vector3> Curve3;
 
     extern template class threepp::Curve<Vector2>;
     extern template class threepp::Curve<Vector3>;

--- a/include/threepp/geometries/TubeGeometry.hpp
+++ b/include/threepp/geometries/TubeGeometry.hpp
@@ -43,7 +43,7 @@ namespace threepp {
                 bool closed = false);
 
     private:
-        Curve3::FrenetFrames frames;
+        FrenetFrames frames;
 
         TubeGeometry(std::shared_ptr<Curve3> path, const Params& params);
     };

--- a/src/threepp/extras/core/Curve.cpp
+++ b/src/threepp/extras/core/Curve.cpp
@@ -190,7 +190,7 @@ void Curve<T>::getTangentAt(float u, T& optionalTarget) const {
     this->getTangent(t, optionalTarget);
 }
 
-Curve3::FrenetFrames Curve3::computeFrenetFrames(unsigned int segments, bool closed) {
+FrenetFrames FrenetFrames::compute(const Curve<Vector3>& curve, unsigned int segments, bool closed) {
 
     // see http://www.cs.indiana.edu/pub/techreports/TR425.pdf
 
@@ -210,7 +210,7 @@ Curve3::FrenetFrames Curve3::computeFrenetFrames(unsigned int segments, bool clo
         const float u = static_cast<float>(i) / static_cast<float>(segments);
 
         auto& tangent = tangents.emplace_back();
-        this->getTangentAt(u, tangent);
+        curve.getTangentAt(u, tangent);
         tangent.normalize();
     }
 

--- a/src/threepp/geometries/ExtrudeGeometry.cpp
+++ b/src/threepp/geometries/ExtrudeGeometry.cpp
@@ -267,7 +267,7 @@ ExtrudeGeometry::ExtrudeGeometry(const std::vector<Shape>& shapes, const Extrude
 
         std::vector<Vector3> extrudePts;
         bool extrudeByPath = false;
-        Curve3::FrenetFrames splineTube;
+        FrenetFrames splineTube;
         Vector3 binormal, normal, position2;
 
         if (extrudePath) {
@@ -281,7 +281,7 @@ ExtrudeGeometry::ExtrudeGeometry(const std::vector<Shape>& shapes, const Extrude
 
             // TODO1 - have a .isClosed in spline?
 
-            splineTube = extrudePath->computeFrenetFrames(steps, false);
+            splineTube = FrenetFrames::compute(*extrudePath, steps, false);
         }
 
         // Safeguards if bevels are not enabled

--- a/src/threepp/geometries/TubeGeometry.cpp
+++ b/src/threepp/geometries/TubeGeometry.cpp
@@ -10,7 +10,7 @@ using namespace threepp;
 TubeGeometry::TubeGeometry(std::shared_ptr<Curve3> path, const Params& params)
     : path(std::move(path)), radius(params.radius) {
 
-    this->frames = this->path->computeFrenetFrames(params.tubularSegments, params.closed);
+    this->frames = FrenetFrames::compute(*this->path, params.tubularSegments, params.closed);
 
     // helper variables
 

